tools/windows-node-installer/test/e2e/common.go

```diff
@@ -0,0 +1,84 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/masterzen/winrm"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/types"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// winRMPort is the secure WinRM port
+	winRMPort = "5986"
+	// winFirewallCmd will verify if firewall rule is present on the windows node
+	winFirewallCmd = "powershell.exe Get-NetFirewallRule -DisplayName ContainerLogsPort"
+)
+
+var (
+	// credentials store the windows instance credential information
+	credentials *types.Credentials
+	// retryCount is the no of times we hit a request.
+	retryCount = 30
+	// retryInterval is interval of time until we retry after a failure.
+	retryInterval = 5 * time.Second
+)
+
+// setupWinRMClient initializes the winRM client and executes commands on the windows node
+func setupWinRMClient(host, password, user string) (*winrm.Client, error) {
+	winRMPortNo, err := strconv.Atoi(winRMPort)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to an integer with error: %v", err)
+	}
+	endpoint := winrm.NewEndpoint(host, winRMPortNo, true, true,
+		nil, nil, nil, time.Minute*1)
+	winRMClient, err := winrm.NewClient(endpoint, user, password)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize winRM client with error: %v", err)
+	}
+	return winRMClient, nil
+}
+
+// runWinRMCmd executes a winRM command on the windows node.
+func runWinRMCmd(winRMClient *winrm.Client, command string) (string, string, error) {
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+	exitCode, err := winRMClient.Run(command, stdout, stderr)
+	if err != nil {
+		return "", "", fmt.Errorf("failed while executing the %s remotely with error: %v", winFirewallCmd, err)
+	}
+	if exitCode != 0 {
+		return stdout.String(), stderr.String(), fmt.Errorf("%s returned %d exit code", winFirewallCmd, exitCode)
+	}
+	return stdout.String(), stderr.String(), nil
+}
+
+// checkForFirewallRule waits until it is able to locate firewall rule on windows node
+// for retryCount attempts
+func checkForFirewallRule(winRMClient *winrm.Client) (stdout, stderr string, err error) {
```

Comment:
What is the point of `stdout`? You are not using it anywhere.

Reply:
yes I don't need to return stdout from checkForFirewallRule. Will update that.